apps/settings/lib/Controller/AppSettingsController.php

```diff
@@ -106,6 +123,84 @@ public function viewApps(): TemplateResponse {
 		return $templateResponse;
 	}
 
+	/**
+	 * Get all active entries for the app discover section
+	 *
+	 * @NoCSRFRequired
+	 */
+	public function getAppDiscoverJSON(): JSONResponse {
+		// $data = $this->discoverFetcher->get();
+
+		// TODO REMOVE JUST FOR TESTING
+		$file = file_get_contents(__DIR__ . '/../../discover.json');
+		$data = json_decode($file);
+		// END TODO
+		return new JSONResponse($data);
+	}
+
+	/**
+	 * @PublicPage
+	 * @NoCSRFRequired
+	 *
+	 * Get a image for the app discover section - this is proxied for privacy and CSP reasons
+	 *
+	 * @param string $image
+	 * @throws \Exception
+	 */
+	public function getAppDiscoverMedia(string $fileName): Response {
+		$etag = $this->discoverFetcher->getETag() ?? date_format(new \DateTimeImmutable(), 'Y-m');
+		try {
+			$folder = $this->appData->getFolder('app-discover-cache');
+			// Cleanup old cache folders
+			$allFiles = $folder->getDirectoryListing();
+			foreach ($allFiles as $dir) {
+				if ($dir->getName() !== $etag) {
+					$dir->delete();
+				}
+			}
+		} catch (NotPermittedException $e) {
+			// can be ignored - this is from the cleanup
+		} catch (\Throwable $e) {
+			$folder = $this->appData->newFolder('app-discover-cache');
+		}
+
+		// Get the current cache folder
+		try {
+			$folder = $folder->getFolder($etag);
+		} catch (NotFoundException $e) {
+			$folder = $folder->newFolder($etag);
+		}
+
+		$info = pathinfo($fileName);
+		$hashName = md5($fileName);
+		$allFiles = $folder->getDirectoryListing();
+		// Try to find the file
+		$file = array_filter($allFiles, function (ISimpleFile $file) use ($hashName) {
+			return str_starts_with($file->getName(), $hashName);
+		});
+		$file = $file[0] ?? null;
```

Comment:
If key `0` if delete by `array_filter`, it will not work. Instead:
```suggestion
		$file = reset($file) ?? null;
```